catalog/admin/module/catalog-admin-module-maplayers/src/main/webapp/map-layers/reducer.js

```diff
@@ -45,7 +47,20 @@ export const fetch = () => (dispatch) => {
     .then((res) => res.json())
     .then((json) => {
       const config = fromJS(json).getIn(configPath)
-        .update('imageryProviders', (providers) => fromJS(JSON.parse(providers || '[]')))
+        .update('imageryProviders', (providers) => {
+          try {
+            const parsed = JSON.parse(providers)
+            const err = validateStructure(parsed)
+            if (err !== undefined) {
+              throw Error(err)
```

Comment:
@lessarderic Talked with @djblue about this, unfortunately because JSON.parse already throws exceptions  (and I would argue throwing more errors by changing validateStructure is not the right path) , the simplest way to deal with this is to have the error check and throw. You could not throw the error and just deal with it inline, but then you get duplicated code and a little deeper nesting. Even with abstracting the error handling to a method it doesn't look much cleaner (Feels like splitting up the related code far apart.. at least without changing some of these to be vars instead of consts) 
```javascript
try {
  const parsed = JSON.parse(providers)
  const err = validateStructure(parsed)
  if (err !== undefined) {
    return emptyAndDispatch(e, providers)
  }
} catch (e) {
  return emptyAndDispatch(e, providers)
}
```